Fixes asString when query has 1 param with no value

albertpastrana opened the pull request:
When there was a key with no values in the query string,
there was a implicit conversion from string to list that
made the code compile but that messed up with the params.

That way, this query string `?hello` would be translated
into `?h&e&l&l&o`.

Signed-off-by: Albert Pastrana <albert.pastrana@gmail.com>

albertpastrana committed:
Fixes asString when query has 1 param with no value

When there was a key with no values in the query string,
there was a implicit conversion from string to list that
made the code compile but that messed up with the params.

That way, this query string `?hello` would be translated
into `?h&e&l&l&o`.

Signed-off-by: Albert Pastrana <albert.pastrana@gmail.com>

albertpastrana commented:
@nathankleyn it would be great if you can take a look at this one :)

nathankleyn commented on url/src/test/scala/uscala/net/URLSpec.scala:
```diff
@@ -69,40 +68,41 @@ class URLSpec extends Specification with ScalaCheck with TestData {
     "should decode the rawPath correctly" >> {
       "when it has a space" >> {
-        URL("http://example.com/hello%20world/").map(_.path) must beASuccessfulTry(Some("/hello world/"))
+        URL("http://example.com/hello%20world/").map(_.path) must beASuccessfulTry(Option("/hello world/"))
```
Think both of these lines should still be `Some`?

albertpastrana replied:
This is because Intellij, if I leave it as `Some`, then the line appears as a compilation error (even if it's not). So I have to either do what I did or leave as `Some` and add the type like this: `beASuccessfulTry[Option[String]](Some("/hello world/"))`.

nathankleyn reviewed (changes requested):
One super quick thing, up to you if you want to fix as it works as it is. 👍

albertpastrana committed:
Fixes exponential backoff test

Can't remember where I got the numbers for the exponential
backoff test, but after several intermitent failures I just
checked what the min and max values should be and they are
slightly different from what I got.

With these new values the test should never fail.

Signed-off-by: Albert Pastrana <albert.pastrana@gmail.com>

nathankleyn reviewed (approved):
LGTM! 👍 👏
